worker/umap.worker.ts

```diff
@@ -0,0 +1,25 @@
+import { UMAP } from '../src/lib';
+
+async function messageHandler({ data: { id, method, args } }) {
+  console.log(
+    `UMAPWorker: received instruction to execute 'umap.${method}' with ${args.length} args`
+  );
+  switch (method) {
+    case 'fit': {
+      const umap = new UMAP();
+      const embedding = await umap.fit(args[0]);
+      // @ts-ignore
```

Comment:
TypeScript is incorrectly seeing the `postMessage`s here as `window.postMessage`s instead of `Worker.postMessage`s and thus wanting a `targetOrigin` 2nd argument. This is almost certainly related to the `@ts-ignore` required to used the worker-loader on the module (see comment below). Getting the Worker to bundle via Webpack with TypeScript was the hardest part of this. I've bundled Workers into libs many times with the worker-loader and not had any problems in plain JS but Googling around confirms that this is indeed a common issue. (I'm also not sure how cross-browser friendly this bundling approach will be, either, since I'm not sure it works without `{ fallback: false, inline: true }` and that may exclude some browsers...)

Reply:
Yeah, I had to deal with this once before and I wound up opting just to redefine a type for postMessage on the window object.... kind of equally hacky but at least you get type hints on that method. As far as the browser compatibility goes I think we can cross that bridge when we get there.